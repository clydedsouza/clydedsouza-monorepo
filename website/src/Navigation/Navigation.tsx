import { Outlet, Link } from "react-router-dom";

function Navigation() {
    return (
      <div>
       <nav>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/projects">Blogs</Link>
          </li>
          <li>
            <Link to="/headline">Healdine</Link>
          </li>
        </ul>
      </nav>

      <Outlet />
      </div>
    );
  }
  
  export default Navigation;