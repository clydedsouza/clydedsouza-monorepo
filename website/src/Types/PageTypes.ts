export enum PageTypes {
  About = 'About',
  Projects = 'Projects',
  Speaking = 'Speaking',
  Teaching = 'Teaching',
  Books = 'Books',
  Highlights = 'Highlights',
  Pinned = 'Pinned',
}
