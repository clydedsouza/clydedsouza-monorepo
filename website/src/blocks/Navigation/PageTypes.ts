export enum PageTypes {
  About = "About",
  Portfolio = "Portfolio",
  Platforms = "Platforms",
  Highlights = "Highlights",
  Cta = "Cta",
  Blog = "Blog",
}
