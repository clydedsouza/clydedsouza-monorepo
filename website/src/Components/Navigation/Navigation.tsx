import { BrowserRouter, NavLink, Outlet, Route, Routes } from 'react-router-dom'
import AboutMe from '../../Pages/AboutMe/AboutMe'
import Projects from '../../Pages/Projects/Projects'
import { PageTypes } from '../../Types/PageTypes'
import './Navigation.scss'

function Menu() {
  return (
    <>
      <nav>
        <NavLink
          to="/"
          className={({ isActive }) => (isActive ? 'active' : '')}
        >
          {PageTypes.About}
        </NavLink>
        <NavLink
          to={PageTypes.Highlights.toLowerCase()}
          className={({ isActive }) => (isActive ? 'active' : '')}
        >
          {PageTypes.Highlights}
        </NavLink>
        <NavLink
          to={PageTypes.Platforms.toLowerCase()}
          className={({ isActive }) => (isActive ? 'active' : '')}
        >
          {PageTypes.Platforms}
        </NavLink>
      </nav>
      <Outlet />
    </>
  )
}

function Navigation() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<Menu />}>
          <Route index element={<AboutMe />} />
          <Route
            path={PageTypes.Highlights.toLowerCase()}
            element={<Projects name={PageTypes.Highlights} />}
          />
          <Route
            path={PageTypes.Platforms.toLowerCase()}
            element={<Projects name={PageTypes.Platforms} />}
          />
          <Route path="*" element={<AboutMe />} />
        </Route>
      </Routes>
    </BrowserRouter>
  )
}

export default Navigation
