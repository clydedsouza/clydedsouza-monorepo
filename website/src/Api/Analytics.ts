import mixpanel from 'mixpanel-browser'
import { PageTypes } from '../Components/Navigation/PageTypes'
import {
  ILinkClickedProps,
  MIXPANEL_DEV_PROJECT_ID,
  MIXPANEL_PROD_PROJECT_ID,
} from './IAnalytics'

export const initAnalyticsWithSuperProperties = () => {
  const mixpanelProjectId =
    process.env.NODE_ENV === 'production'
      ? MIXPANEL_PROD_PROJECT_ID
      : MIXPANEL_DEV_PROJECT_ID

  try {
    mixpanel.init(mixpanelProjectId)
    mixpanel.register({
      environment: process.env.NODE_ENV,
    })
  } catch {
    undefined
  }
}

export const sendPageViewEvent = (pageType: PageTypes) => {
  try {
    mixpanel.track('Page view', { page: pageType })
  } catch {
    undefined
  }
}

export const sendLinkClickedEvent = (props: ILinkClickedProps) => {
  try {
    mixpanel.track('Link clicked', {
      destination_url: props.link,
      link_type: props.type,
      page_location: props.location,
    })
  } catch {
    undefined
  }
}
