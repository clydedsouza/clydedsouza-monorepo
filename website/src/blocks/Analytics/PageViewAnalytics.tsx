"use client";

import { sendPageViewEvent } from "@/api/Analytics";
import { useEffect } from "react";
import { PageTypes } from "../Navigation/PageTypes";

export const PageViewAnalytics: React.FC<{ page: PageTypes }> = ({ page }) => {
  useEffect(() => {
    sendPageViewEvent(page);
  }, []);
  return <></>;
};
