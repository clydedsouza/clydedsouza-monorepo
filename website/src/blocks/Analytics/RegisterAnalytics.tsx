"use client";

import { initAnalyticsWithSuperProperties } from "@/api/Analytics";
import { useEffect } from "react";

export const RegisterAnalytics = () => {
  useEffect(() => {
    initAnalyticsWithSuperProperties();
  }, []);
  return <></>;
};
