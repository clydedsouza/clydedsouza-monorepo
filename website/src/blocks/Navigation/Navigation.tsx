"use client";

import { sendNavigationClickedEvent } from "@/api/Analytics";
import Link from "next/link";
import { usePathname } from "next/navigation";
import { useState } from "react";
import "./Navigation.scss";
import { PageTypes } from "./PageTypes";

const getCurrentPage = (pathFromNextJs: string) => {
  const cleanPath = pathFromNextJs.replaceAll("/", "").toLowerCase();

  if (cleanPath === PageTypes.Platforms.toLowerCase()) {
    return PageTypes.Platforms;
  } else if (cleanPath === PageTypes.Portfolio.toLowerCase()) {
    return PageTypes.Portfolio;
  } else {
    return PageTypes.About;
  }
};

function Navigation() {
  const [currentPage, setCurrentPage] = useState(getCurrentPage(usePathname()));

  const onNavigationItemClick = (pageType: PageTypes) => {
    sendNavigationClickedEvent(pageType);
    setCurrentPage(pageType);
  };

  return (
    <>
      <nav>
        <Link
          href="/"
          className={currentPage === PageTypes.About ? "active" : ""}
          onClick={() => onNavigationItemClick(PageTypes.About)}
        >
          {PageTypes.About}
        </Link>
        <Link
          href={PageTypes.Portfolio.toLowerCase()}
          className={currentPage === PageTypes.Portfolio ? "active" : ""}
          onClick={() => onNavigationItemClick(PageTypes.Portfolio)}
        >
          {PageTypes.Portfolio}
        </Link>
        <Link
          href={PageTypes.Platforms.toLowerCase()}
          className={currentPage === PageTypes.Platforms ? "active" : ""}
          onClick={() => onNavigationItemClick(PageTypes.Platforms)}
        >
          {PageTypes.Platforms}
        </Link>
        <Link
          href={"https://blog.clydedsouza.net"}
          target="_blank"
          onClick={() => sendNavigationClickedEvent(PageTypes.Blog)}
        >
          {PageTypes.Blog}
        </Link>
      </nav>
    </>
  );
}

export default Navigation;
