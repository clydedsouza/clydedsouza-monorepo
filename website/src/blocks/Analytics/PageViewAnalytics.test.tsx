import * as analyticsModule from "@/api/Analytics";
import { render } from "@testing-library/react";
import { PageTypes } from "../Navigation/PageTypes";
import { PageViewAnalytics } from "./PageViewAnalytics";

jest.mock("../../api/Analytics");

describe("Page View Analytics", () => {
  it("should render component", () => {
    const { container } = render(
      <PageViewAnalytics page={PageTypes.Platforms} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should initialize analytics when component is rendered", async () => {
    const sendPageViewEventMock = jest
      .spyOn(analyticsModule, "sendPageViewEvent")
      .mockImplementation(jest.fn());

    render(<PageViewAnalytics page={PageTypes.Platforms} />);

    expect(sendPageViewEventMock).toHaveBeenCalledWith(PageTypes.Platforms);
  });
});
