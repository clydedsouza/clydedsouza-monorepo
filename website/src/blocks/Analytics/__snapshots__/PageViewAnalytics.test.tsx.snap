// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page View Analytics should render component 1`] = `<div />`;
